Makefile

```diff
@@ -26,77 +25,171 @@ export PRINT_HELP_PYSCRIPT
 	rm -fr dist/
 	rm -fr .eggs/
 	find . -name '*.egg-info' -exec rm -fr {} +
 	find . -name '*.egg' -exec rm -f {} +
 
+.PHONY: clean-pyc
 clean-pyc: ## remove Python file artifacts
 	find . -name '*.pyc' -exec rm -f {} +
 	find . -name '*.pyo' -exec rm -f {} +
 	find . -name '*~' -exec rm -f {} +
 	find . -name '__pycache__' -exec rm -fr {} +
 
+.PHONY: clean-docs
+clean-docs: ## remove previously built docs
+	-$(MAKE) -C docs clean 2>/dev/null  # this fails if sphinx is not yet installed
+
+.PHONY: clean-coverage
 clean-coverage: ## remove coverage artifacts
 	rm -f .coverage
 	rm -f .coverage.*
 	rm -fr htmlcov/
 
-clean-test: ## remove test and coverage artifacts
+.PHONY: clean-test
+clean-test: ## remove test artifacts
 	rm -fr .tox/
 	rm -fr .pytest_cache
 
+.PHONY: clean
+clean: clean-build clean-pyc clean-test clean-coverage clean-docs ## remove all build, test, coverage, docs and Python artifacts
+
+
+# INSTALL TARGETS
+
+.PHONY: install
+install: clean-build clean-pyc ## install the package to the active Python's site-packages
+	pip install .
+
+.PHONY: install-test
+install-test: clean-build clean-pyc ## install the package and test dependencies
+	pip install .[test]
+
+.PHONY: install-develop
+install-develop: clean-build clean-pyc ## install the package in editable mode and dependencies for development
+	pip install -e .[dev]
+
+
+# LINT TARGETS
+
+.PHONY: lint
 lint: ## check style with flake8 and isort
-	flake8 btb tests examples
-	isort -c --recursive btb tests examples
+	flake8 btb tests
+	isort -c --recursive btb tests
 
+.PHONY: fix-lint
 fix-lint: ## fix lint issues using autoflake, autopep8, and isort
-	autopep8 --in-place --recursive --aggressive --aggressive btb tests examples
-	isort --apply --atomic --recursive btb tests examples
+	find btb -name '*.py' | xargs autoflake --in-place --remove-all-unused-imports --remove-unused-variables
+	autopep8 --in-place --recursive --aggressive btb
+	isort --apply --atomic --recursive btb
 
+	find tests -name '*.py' | xargs autoflake --in-place --remove-all-unused-imports --remove-unused-variables
+	autopep8 --in-place --recursive --aggressive tests
+	isort --apply --atomic --recursive tests
+
+
+# TEST TARGETS
+
+.PHONY: test
 test: ## run tests quickly with the default Python
-	pytest
+	python -m pytest --cov=btb
 
+.PHONY: test-all
 test-all: ## run tests on every Python version with tox
 	tox
 
-coverage: clean-coverage ## check code coverage quickly with the default Python
+.PHONY: coverage
+coverage: ## check code coverage quickly with the default Python
 	coverage run --source btb -m pytest
 	coverage report -m
 	coverage html
 	$(BROWSER) htmlcov/index.html
 
-clean-docs: ## remove previously built docs
-	rm -f docs/api/*.rst
-	$(MAKE) -C docs clean
 
+# DOCS TARGETS
+
+.PHONY: docs
 docs: clean-docs ## generate Sphinx HTML documentation, including API docs
-	sphinx-apidoc --module-first --separate -o docs/api/ btb
 	$(MAKE) -C docs html
+	touch docs/_build/html/.nojekyll
```

Comment:
not needed. we already rely on `sphinx.ext.githubpages` which creates the `.nojekyll` file.

Reply:
removing